learningequality/studio
# Remove the tree viewset

## Description

* The relatively straight forward proposition of this PR is to remove the tree viewset and just go back to querying contentnodes
* This is not so straight forward as much of the frontend and backend code is built predicated on this separation
* Refactor all frontend ContentNode vuex state to just work directly with ContentNodes and not separate tree node representations
* Update all copy operations for ContentNodes in the tree to work on just a ContentNode
* Update resource loading on the channel edit page
* Switch caching strategy from matching indexedDB queries to parameter specific in order to prevent synchronous requests from hitting the backend twice.
* Update the clipboard to no longer rely on the Tree
* Create a separate Clipboard viewset to handle incremental loading of the clipboard and node creation within the clipboard
* Update the clipboard copy strategy to only copy children when they differ from the original node children
* Dynamically load children in the clipboard depending on whether it is a deep representation
* Add logging messages in the resource layer to allow for easier stack traces to understand where async calls originate
* Update the sync code to sync more frequently


#### Issue Addressed (if applicable)

Fixes #2105
Fixes #1941
Fixes #1882


## Steps to Test

* Check that channels with large trees load channel edit
* Check that you can copy content nodes (caveat - assessmentitem file copying is currently broken from a pre-existing issue)
* Check that you can copy content nodes to and from the clipboard
